### PR TITLE
Data checks

### PR DESCRIPTION
**Pull request by vaelliott**
Ready for another cross-examination.

I've responded to your comments and hopefully made some improvements - there are still a couple of things that aren't quite right (i'll email you the details).

I see there are merge conflicts but I expect that is just to do with the age_group function that is now in master.  I don't think there should be anything too tricky to sort out.

### CODE REVIEW

**Comment by vaelliott** [comment]
hi jaime
I've made a few more changes - namely to make the qa function run at the end of analyse_second.  Hopefully it will automatically detect which organisation is running the code as my report won't work for HS.  I have decided after much annoyance that the report still occasionally has issues but I think its to do with pandoc versions but this shouldn't stop the function from being usable as it is.

**Comment by jvillacampa** [comment]
I have checked it finally. It works fine, so I am going to merge it although I have some comments that you might want look at in the future.

- Why the report won't work for HS? As far as I can see, it should. the filepath "data_folder" already depends on the organisation and I can't see any hard coded filepath that would cause a problem. 

- Data check 5 - I got names instead of codes for the chart.

- ABIs which is one of the odd cases works fine for me. However for data check 4 it says: "Uh-oh looks like there are some rates outwith confidence limits :(" and it could just say that there are no CIs which is the case.

- I am dealing with the merge conflicts.

- I am deleting the test file. I'll save a copy and send it to you just in case, but I don't see the point on keeping it.

- Are you planning to adapt it for the deprivation analysis?

**Comment by jvillacampa** [comment]
Also, the rmd file could be renamed to 3. Data checks or similar, so it's not lost among all the indicator code

File: Data Quality Checks.Rmd [hunk 1]
```diff
@@ -13,90 +11,101 @@ knitr::opts_chunk$set(echo = TRUE)
...
+code_dictionary <- readRDS(paste0(data_folder,"Lookups/Geography/codedictionary.rds")) 
 
 #Read in indicator file being checked
 data_indicator <- readRDS(paste0(data_folder, "Data to be checked/", filename, "_shiny.rds")) 
 
+#Add geography code names
+data_indicator <- left_join(x=data_indicator, y=code_dictionary, by="code")
+
+#Test which optional geographies are present in dataset
+board <- ifelse(as.numeric(count(data_indicator %>% filter(substr(code,1,3) == "S08"))) > 0,TRUE, FALSE)
```
Comment (jvillacampa) [inline Data Quality Checks.Rmd]: These tests could be simplified using any (all is also an useful function):
any(substr(data_indicator$code,1,3) == "S08")

Comment (jvillacampa) [reply Data Quality Checks.Rmd]: Done

File: Data Quality Checks.Rmd [hunk 2]
```diff
@@ -202,137 +245,194 @@ scot_check
...
+  summarise(numerator=sum(numerator)) %>%
+  spread(geotype, numerator) %>%
+  mutate(ca_partnership=ifelse((hscp==TRUE && ca==TRUE),(round(S12-S37,2)),0),  #create fields that test for differences - use rounded figures
+         partnership_localities=ifelse(locality==TRUE,(round(S99-S37,2)),0), #if dataset includes localities check sum
+         localities_ca=ifelse(locality==TRUE,(round(S99-S12,2)),0),
+         check_tot=round(sum(ca_partnership,partnership_localities,localities_ca),2))}
+
+if(all(data_indicator$numerator == "")){
+  ed_flex <- "No numerator to cross reference totals"
+} else if (sum(edinburgh_data$check_tot) !=0){
```
Comment (jvillacampa) [inline Data Quality Checks.Rmd]: I have added a na.rm=T to the sum as it was causing problems when running the access deprived indicator (as for some years there was no data for some geographies

**Pull request merged**